### ISSUE TITLE
Invalid interop compatibility

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtimelab#168]
Interop/PInvoke/Array/MarshalArrayAsField/AsLPArray/AsLPArrayTest

```
Test FAIL: TestLibrary.AssertTestException: Assert.Fail: Expected 'System.TypeLoadException' to be thrown, however 'System.Exception' was thrown. TakeIntArraySeqStructByVal
   at TestLibrary.Assert.HandleFail(String, String) + 0x5a
   at TestLibrary.Assert.Fail(String, Object[]) + 0x64
   at TestLibrary.Assert.Throws[T](Action, String, AssertThrowsOptions) + 0x1e1
   at Test.RunTest1(String) + 0xd4
   at Test.Main(String[]) + 0x29
```

**Comment by kant2002** [comment]
What should be done here for HandleRef support? Usual stuff for Marshallers? Or any caveats?

**Comment by MichalStrehovsky** [comment]
Yeah, I think HandleRef is the usual stuff. I think we already support it. Maybe not fully?

This issue specifically tracks that if the HandleRef is invalid, we don't throw a descriptive exception. (`"Expected 'System.Runtime.InteropServices.MarshalDirectiveException' to be thrown, however 'System.Exception' was thrown."`).

**Comment by MichalStrehovsky** [comment]
(I think the issue you're seeing with CreateStdAccessibleObject is caused by the object parameter. We don't have handling for other NativeType than AsAny - we need this to take the COM marshaller. Hopefully a single-line fix:

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs#L556-L559

**Comment by kant2002** [comment]
Thanks for clarification. If one liner fix issues thats would be quite good.
